os/drivers/lcd/mipi_lcd.c

```diff
@@ -399,6 +405,38 @@ static int lcd_setcontrast(FAR struct lcd_dev_s *dev, unsigned int contrast)
 	return OK;
 }
 
+FAR void lcd_set_logo(FAR struct lcd_dev_s *dev)
+{
+	int logo_arr_index = 0;
+	int lcd_data_index = CONFIG_LCD_XRES * (CONFIG_LCD_YRES - CONFIG_LCD_LOGO_YRES) + (CONFIG_LCD_XRES - CONFIG_LCD_LOGO_XRES);
+	int lcd_data_col_count = 0;
+	FAR struct mipi_lcd_dev_s *priv = (FAR struct mipi_lcd_dev_s *)dev;
+
+#if !defined(CONFIG_LCD_SW_ROTATION)
+	lcd_data_logo = (uint8_t *)kmm_malloc(CONFIG_LCD_XRES * CONFIG_LCD_YRES * 2 + 1);
+	if (!lcd_data_logo) {
+		lcddbg("ERROR: LCD logo data memory allocation failed\n");
+		return;
+	}
+#endif
+	for (int i = 0; i < CONFIG_LCD_XRES * CONFIG_LCD_YRES * 2; i++) {
+		lcd_data_logo[i] = 0;
+	}
+
+	while (logo_arr_index <= (CONFIG_LCD_LOGO_YRES * CONFIG_LCD_LOGO_XRES * 2)) {
+		lcd_data_logo[lcd_data_index] = lcd_logo_data[logo_arr_index++];
+		lcd_data_logo[lcd_data_index + 1] = lcd_logo_data[logo_arr_index++];
```

Comment:
lcd_data_logo is for sw rotation??

Reply:
I have formatted the code and added comment
https://github.com/Samsung/TizenRT/pull/6521/files#:~:text=/*%20Memory%20optimization%20applied,*/

Reply:
it's not used for sw rotation.
the rotation buffer is idle at the time of bootup. it's not being used. so, instead of creating and allocating new buffer of lcd size, we are just using the rotation buffer.